plugins/MauticCrmBundle/Integration/DynamicsIntegration.php

```diff
@@ -629,8 +689,8 @@ public function amendLeadDataBeforeMauticPopulate($data, $object = null)
                         $result[] = $entity->getName();
                     }
                     $mauticObjectReference = 'company';
-                } elseif ('contacts' === $object) {
-                    $recordId = $entityData['contactid'];
+                } elseif ('contacts' === $object || 'leads' === $object) {
+                    $recordId = ('contacts' === $object) ? $entityData['contactid'] : $entityData['leadid'];
                     // first try to find integration entity
                     $integrationId = $integrationEntityRepo->getIntegrationsEntityId('Dynamics', $object, 'lead',
```

Comment:
Hi @npracht, 
I found something very strange that is preventing records by updated from Dynamics:
This line returns nothing but If check what is set to be returned from inside getIntegrationsEntityId() ir seems to be OK.

Using PHP 7.2.19-0ubuntu0.18.04.1 (cli) (built: Jun  4 2019 14:48:12) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.19-0ubuntu0.18.04.1, Copyright (c) 1999-2018, by Zend Technologies